BUILD.pants

```diff
@@ -0,0 +1,38 @@
+# This target is what invokes Bazel. The main point to note here is that the `{chroot}` symbol is replaced with the path to the temporary directory used during execution to materialize Pants-supplied inputs. 
+shell_command(
+  name="bazel-jvm-binary",
+  command="bazel build --java_runtime_version=remotejdk_11 :ProjectRunner && cp bazel-bin/ProjectRunner.jar {chroot} && ls {chroot}",
+  tools=["bazel", "gcc", "cp", "ls"],
+  extra_env_vars=["HOME", "PATH"],
+  workdir="bazel-jvm",
+  output_files=["ProjectRunner.jar"],
+
+  # This field sets what environment to execute the command in. The special value
+  # `__local_workspace__` selects whatever configured `experimental_workspace_environment`
+  # matches the current platform.
+  environment="__local_workspace__",
+
+  # The `path_env_modify` field allows you to disable modification of the PATH environment
+  # variable by Pants. Since Bazel may incorporate the exact PATH value into its cache
+  # key, this reduces the non-deteriminsm to invoking Bazel to allow Bazel to use its 
+  # cache more effectively.
+  path_env_modify="off",
+
+  # Configure a longer timeout to give Bazel a chance to complete when necessary.
+  timeout=600,
+)
+
+# For the sake of this example, we will just output the `ls -l` output to demonstrate
+# that the jar built by Bazel is visible to Pants.
+run_shell_command(
```

Comment:
People might wonder how to use this jar as a resource, say to embed it in a docker image.  Is that straightforward?

Also, instead of `ls` maybe we should at least demonstrate running the code in the jar?

Reply:
Do we have a way to easily expose a Coursier-managed JVM to shell / adhoc target types?

Reply:
My thought being: If we want to execute the jar, then we need a JVM available, hopefully easily.

Reply:
Never mind, went with `system_binary` to find Coursier. (The earlier thought was that we have a target type which exposes the exact Coursier used by Pants for the JVM backend. But this example does not use the Pants JVM backends, so that would not make for a good example in any event.)

Reply:
@benjyw: Any thoughts on the change to just run the jar with Coursier?

Reply:
Sure, that could make sense, I see what you mean about not using the JVM backends.

Reply:
Although, that said, maybe it should use the JVM backend, to demonstrate how to consume something built via workspace execution?

Reply:
How would I accomplish that with the JVM backend as is? I'm not aware of a "wrap as" target or equivalent for jars.

Reply:
I don't know, I was hoping to see that demonstrated :) 

OK, so how about using the docker backend to embed the jar in an image? Something to show how you can take the result of this and use it upstream in a "regular" pants build?

Reply:
That's an interesting idea. I'll see if I can get that to work.